### ISSUE TITLE
Add possibility to use envrionment variables for different login methods

### ISSUE THREAD

**Issue by DaniRieker** [akeyless-community/terraform-provider-akeyless#53]
It would be useful to have the possibility of using different login methods on the local machine than inside pipelines. By using different environment variables to determine the login method and doing the authentication we would be able to use the personal identity while developing on the local machine and an api_key_login when running inside pipelines. The provider config would not contain any login information in this case and the provider would than check for the existence of environment variables to do the login.

For example 
- AKEYLESS_AUTH_ACCESS_ID & AKEYLESS_AUTH_ACCES_KEY
- AKEYLESS_AUTH_AZUREAD
- AKEYLESS_AUTH_AWS
- AKEYLESS_AUTH_JWT
- AKEYLESS_AUTH_ADMIN_EMAIL & AKEYLESS_AUTH_ADMIN_PASSWORD

For JWT_LOGIN this option already exists. As described in the docs: https://registry.terraform.io/providers/akeyless-community/akeyless/latest/docs#schema

**Comment by devorbitus** [comment]
You can already do this by utilizing the built-in override files.
https://developer.hashicorp.com/terraform/language/files/override

You just need to add some override files to your .gitignore file then locally, you can create authentication more suitable for local development like SAML or other SSO requiring a browser.

The way it works is you can have something like a `normal-backend.tf` that gets checked into the repo but then a local `override.tf` file that, when exists, will be used instead of the conflicting `normal-backend.tf` configuration.

**Comment by DaniRieker** [comment]
Thanks for your answer, I will close the Issue 👍

**Comment by DaniRieker** [comment]
We've worked now a bit more with the provider and also used the override files, still I think it would be a better Solution to have this implemented.

For example using ACCESS_ID and ACCESS_SECRET from ENV variables forces us now to create some Input variables in order to assign those values correctly. If working locally without those variables we either need to provide dummy values which is a pain or set a default which would not fail the pipeline anymore if the ENV  variables are not set.

Also having a separate override file in each an every repo just locally is a lot of additional work which could be avoided.

For all the other terraform providers we work with, this mechanism is there which shows that this is a way that is heavily used to be more flexible.

**Comment by devorbitus** [comment]
I agree, I had already created a ticket in our backlog to make the development experience better. 

I would prefer if when doing local development, if there was an override environment variable that would allow the provider to function differently for local development rather than needing to change the actual code to run in the CI/CD but only AFTER performing local development of the Terraform IaC.

**Comment by DaniRieker** [comment]
While having a look into this again, I've also looked into the source code and found out that for many login methods the option to use ENV-variables already exist:
AKEYLESS_ACCESS_ID
AKEYLESS_ACCESS_KEY
AKEYLESS_AUTH_JWT
AKEYLESS_EMAIL
AKEYLESS_PASSWORD
AKEYLESS_AUTH_UID
AKEYLESS_AUTH_CERT
AKEYLESS_AUTH_KEY

Found here: https://github.com/akeyless-community/terraform-provider-akeyless/blob/709479438702593429b5ac045d2d19ffbf1adf18/akeyless/provider.go#L34

So only the docs of the provider is missing this information... 
I guess it shouldn't be necessary to check the go-code for getting to know how to set those variables

**Comment by akljph** [comment]
Hi all. Is there any work that still needs to be done or can we close this issue?

**Comment by sa-p-cpm-logicapp** [comment]
From my point fo view there ar two things to be done:

- Add documentation for the existing ENV-Variables
- Implement the usage of ENV-Variables for configuring the login methods that doesn't support this yet (aws_iam, azure_ad, gcp,  uid) and update the documentation as well

**Comment by DaniRieker** [comment]
From my point fo view there ar two things to be done:

- Add documentation for the existing ENV-Variables
- Implement the usage of ENV-Variables for configuring the login methods that doesn't support this yet (aws_iam, azure_ad, gcp,  uid) and update the documentation as well
